### ISSUE TITLE
Can I override existing associations?

### ISSUE THREAD

**Issue by PlanetWilson** [whomwah/qlstephen#57]
macOS seems to already have an association for .ts files which is as video. I wish to have it as text, any way to do that with QLStephen?

**Comment by tsdorsey** [comment]
Yes and no.

tl;dr If a file a text based file and no quicklook plugin is more specific than qlstephen about what type of file they can render qlstephen will render it.

Long answer.
Apply has built a [hierarchical system of file types](https://developer.apple.com/library/content/documentation/General/Conceptual/DevPedia-CocoaCore/UniformTypeIdentifier.html) and it registers many of these at the [system level](https://developer.apple.com/library/content/documentation/Miscellaneous/Reference/UTIRef/Articles/System-DeclaredUniformTypeIdentifiers.html#//apple_ref/doc/uid/TP40009259-SW1)

qlstephen [uses these types](https://github.com/whomwah/qlstephen/blob/9cbf597dba0cbb790b0d76d10a22e0ad56b0f23a/QuickLookStephenProject/QLSFileAttributes.m#L99) to figure out if it can render a given file. If the UTI type has text in the name or is kUTTypeText it will render it.

qlstephen registers itself as a handler of the base UTI type public.data. This means that macOS will always fallback to qlstephen for rendering if nothing else says yes first. It also means if any quick look plugin registers itself as a hander farther down the chain that plugin will take precedence.

So the conditions for qlstephen to render a file are:
- Must have `text` in the UTI mime type
- Must not have a quicklook plugin "higher up the foodchain" that is taking ownership

**Comment by Kymer** [comment]
I have a similar issue but the QuickLook plugin that is causing the issue is not "higher up in the foodchain", it has also registered itself as a handler of the base UTI type public.data.

```
qlmanage -m plugins | grep public.data
  public.data -> /Applications/krita.app/Contents/Library/QuickLook/kritaquicklook.qlgenerator (1 - loaded)
```

Besides deleting the [Krita](https://krita.org) app, how can I make qlstephen take precedence over Krita?

**Comment by Kymer** [comment]
@tstsdorsey Thank you for the quick reply! It's not ideal but it does work.
